sdk/helper/scaleutils/nodepool/combined.go

```diff
@@ -0,0 +1,69 @@
+package nodepool
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+)
+
+const (
+	// CombinedClusterPoolIdentifierAnd requires all identifiers to return true.
+	CombinedClusterPoolIdentifierAnd CombinedClusterPoolIdentifierMode = "and"
+
+	// CombinedClusterPoolIdentifierOr requires at least one identifier to
+	// return true.
+	CombinedClusterPoolIdentifierOr CombinedClusterPoolIdentifierMode = "or"
+)
+
+// CombinedClusterPoolIdentifierMode defines how different
+// ClusterNodePoolIdentifiers are combined.
+type CombinedClusterPoolIdentifierMode string
+
+// combinedClusterPoolIdentifier is an implementation of the
+// ClusterNodePoolIdentifier interface that filters Nomad nodes by combining
+// multiple filters.
+type combinedClusterPoolIdentifier struct {
+	poolIdentifiers []ClusterNodePoolIdentifier
+	mode            CombinedClusterPoolIdentifierMode
+}
+
+// NewCombinedClusterPoolIdentifier returns a new combinedClusterPoolIdentifier.
+func NewCombinedClusterPoolIdentifier(poolIdentifiers []ClusterNodePoolIdentifier, mode CombinedClusterPoolIdentifierMode) ClusterNodePoolIdentifier {
+	return &combinedClusterPoolIdentifier{
+		poolIdentifiers: poolIdentifiers,
+		mode:            mode,
+	}
+}
+
+// IsPoolMember satisfies the IsPoolMember function on the
+// ClusterNodePoolIdentifier interface.
+func (c *combinedClusterPoolIdentifier) IsPoolMember(n *api.NodeListStub) bool {
+	// If mode is 'and' we assume the node is a member unless told otherwise.
+	isMember := c.mode == CombinedClusterPoolIdentifierAnd
+
+	for _, identifier := range c.poolIdentifiers {
+		switch c.mode {
+		case CombinedClusterPoolIdentifierAnd:
+			isMember = isMember && identifier.IsPoolMember(n)
+		case CombinedClusterPoolIdentifierOr:
+			isMember = isMember || identifier.IsPoolMember(n)
+		}
+	}
+	return isMember
+}
+
+// Key satisfies the Key function on the ClusterNodePoolIdentifier interface.
+func (c *combinedClusterPoolIdentifier) Key() string {
+	return "combined_identifier"
+}
+
+// Value satisfies the Value function on the ClusterNodePoolIdentifier
+// interface.
+func (c *combinedClusterPoolIdentifier) Value() string {
+	values := make([]string, 0, len(c.poolIdentifiers))
+	for _, identifier := range c.poolIdentifiers {
+		values = append(values, fmt.Sprintf("%s is %s", identifier.Key(), identifier.Value()))
+	}
+	return strings.Join(values, fmt.Sprintf(" %s ", c.mode))
+}
```

Comment:
I wasn't sure how this should look like. It seems to only be used for log output, so I opted for a more human-friendly representation:

```
2021-11-12T10:13:17.228-0500 [INFO]  policy_eval.worker: scaling target: id=a21d3449-cf30-680b-add1-d1cc4f56616d policy_id=0501a099-feb7-99b7-3b2c-6daa7baa9567 queue=cluster target=aws-asg from=1 to=0 reason="scaling down because metric is 0" meta=map[nomad_policy_id:0501a099-feb7-99b7-3b2c-6daa7baa9567]
2021-11-12T10:13:17.347-0500 [DEBUG] internal_plugin.aws-asg: performing node pool filtering: combined_identifier="node_class is learn and datacenter is batch_workers"
2021-11-12T10:13:17.409-0500 [DEBUG] internal_plugin.aws-asg: performing node selection: selector_strategy=empty_ignore_system
```

I thought of just using `node_class=class and datacenter=dc1`, but that could make parsing harder since `=` is also used as the separator for the other labels. Maybe `node_class:class and datacenter:dc1`?

Reply:
Yeah this is a special case and agree using the equals is probably going to get in the way. I personally find using the colon a little easier to grok but this is very subjective and I think it's fine like it is if you're happy!

Reply:
I'm good either way, so colon it is!